### PR TITLE
Deterministic test ordering [TG-9474]

### PR DESCRIPTION
**Pull request by adamkowalczyk**
Adds a secondary sort on the unique test id to test ordering, to ensure that the ordering is deterministic.

In its current form the `cover-client` integration tests give different results on Node 8/10 and Node 12 for some tests with identical first line numbers.

### CODE REVIEW

File: src/testCombiner.ts [hunk 1]
```diff
@@ -109,14 +109,14 @@ function firstLine(test: ITestData): number {
...
- * Sorts by filename number
+ * Sorts by filename number, then id
  *
  * @param {string} a
  * @param {string} b
  * @returns {number}
  */
 export function cmpLines(a: ITestData, b: ITestData) {
-  return cmp(firstLine(a), firstLine(b));
+  return cmp(firstLine(a), firstLine(b)) || cmp(a.id, b.id);
```
Comment (rjmunro) [inline src/testCombiner.ts]: Why ID and not name?

File: package.json [hunk 2]
```diff
@@ -62,6 +63,7 @@
     "@types/fs-extra": "8.0.0",
     "@types/lodash": "4.14.138",
     "@types/mocha": "5.2.7",
+    "common-tags": "2.0.0-alpha.1",
```
Comment (rjmunro) [inline package.json]: You should put this stuff in it's own PR.

File: src/testCombiner.ts (hunk 1 above)
Comment (adamkowalczyk) [reply src/testCombiner.ts]: The secondary ordering needs to be on a unique value, and the `id`  is guaranteed to be unique. The `name` has not been deduplicated yet, so is not yet guaranteed to be unique.
